docs/dev/modules.md

```diff
@@ -0,0 +1,105 @@
+# Cirq modules 
+
+Cirq has a modular architecture and is organized in a monorepo, all of the modules are in the same folder structure.
+Each module is structured as follows: 
+
+```
+cirq-<module-name>
+   <top level package 1> 
+   <top level package 2>
```

Comment:
I get that there may be multiple of these, but maybe lead with the typical case of one top-level package with name `cirq_<module-name>`

Reply:
add an overview of the structure with the json infrastructure as well? either as part of this or down below by the json stuff if you think that would clutter this first section

Reply:
Sounds good, I added the JSON structure here.